packages/web/src/components/nav/desktop/PlaylistLibrary.tsx

```diff
@@ -90,6 +102,110 @@ const DraggableNavLink = ({
   )
 }
 
+const FolderNavLink = ({
+  id,
+  name,
+  onReorder,
+  children,
+  className,
+  ...buttonProps
+}: PlaylistFolderNavButtonProps) => {
+  const [isDragging, setIsDragging] = useState(false)
+  const onDrag = useCallback(() => {
+    setIsDragging(true)
+  }, [setIsDragging])
+  const onDrop = useCallback(() => {
+    setIsDragging(false)
+  }, [setIsDragging])
+  return (
+    <Droppable
+      key={id}
+      className={styles.droppable}
+      hoverClassName={styles.droppableHover}
+      onDrop={(id: ID | SmartCollectionVariant) => onReorder()}
+      acceptedKinds={['library-playlist', 'playlist-folder']}
+    >
+      <Draggable
+        id={id}
+        text={name}
+        kind='playlist-folder'
+        onDrag={onDrag}
+        onDrop={onDrop}
+      >
+        <button
+          {...buttonProps}
+          draggable={false}
+          className={cn(className, styles.navLink, {
+            [styles.dragging]: isDragging
+          })}
+        >
+          {children}
+        </button>
+      </Draggable>
+    </Droppable>
+  )
+}
+
+const PlaylistFolderNavItem = ({
+  folder,
+  hasUpdate = false,
+  dragging,
+  draggingKind
+}: {
+  folder: PlaylistLibraryFolder
+  hasUpdate: boolean
+  dragging: boolean
+  draggingKind: string
+}) => {
+  const { id, name, contents } = folder
+  return (
+    <Droppable
+      key={id}
+      className={navColumnStyles.droppable}
+      hoverClassName={navColumnStyles.droppableHover}
+      onDrop={() => {}}
+      acceptedKinds={['library-playlist']}
+    >
+      <FolderNavLink
+        id={id}
+        name={name}
+        onReorder={() => {}}
+        className={cn(navColumnStyles.link, {
+          [navColumnStyles.droppableLink]:
+            dragging &&
+            (draggingKind === 'track' ||
```

Comment:
nit: could make a lil helper const for this logic since you reuse it for `disabledLink`, eg `const isDroppableKind = (draggingKind === 'track' || ...`